apps/AEPSampleApp/android/app/src/main/java/com/aepsampleapp/MainApplication.java

```diff
@@ -119,28 +119,21 @@ public void onCreate() {
       } catch (InvocationTargetException e) {
         e.printStackTrace();
       }
-
-      try {
-        Edge.registerExtension();
-        Identity.registerExtension();
-        Messaging.registerExtension();
-        UserProfile.registerExtension();
-        Lifecycle.registerExtension();
-        Signal.registerExtension();
-        Consent.registerExtension();
-        Assurance.registerExtension();
-        Places.registerExtension();
-        Target.registerExtension();
-        Optimize.registerExtension();
-      } catch (InvalidInitException e) {
-        e.printStackTrace();
-      }
       MobileCore.configureWithAppID("YOUR-APP-ID");
-      MobileCore.start(new AdobeCallback() {
-        @Override
-        public void call(Object o) {
-          MobileCore.lifecycleStart(null);
-        }
+      List<Class<? extends Extension>> extensions = Arrays.asList(
+          Lifecycle.EXTENSION,
+          Signal.EXTENSION,
+          Edge.EXTENSION,
+          com.adobe.marketing.mobile.edge.identity.Identity.EXTENSION,
+          Messaging.EXTENSION,
+          UserProfile.EXTENSION,
+          Assurance.EXTENSION,
+          Places.EXTENSION,
+          Target.EXTENSION,
+          Optimize.EXTENSION,
+          Consent.EXTENSION);
+      MobileCore.registerExtensions(extensions, o -> {
+        MobileCore.lifecycleStart(null);
```

Comment:
I don't think we need to have lifecycleStart here.  We can start the lifecycle in onResume().  Still Verifying.

Reply:
Or we can merge this PR for now. And update it in the later PRs if needed.

Reply:
We'd better call lifecycleStart in both places to be safe. I will add a lifecycleStart call in onResume() to avoid misunderstanding.